### PR TITLE
zinc: update to 0.3.15, use https

### PR DESCRIPTION
**Pull request by l2dy**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.6 17G65
Xcode 9.4.1 9F2000

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?

- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/zinc/Portfile [hunk 1]
```diff
@@ -12,15 +12,16 @@ long_description \
...
 
 use_configure   no
 build           {}
 
 extract.suffix  .tgz
-checksums       md5    a1e71c996bfc875acaa32ea192ef8df8 \
-                sha1   ff91434ce32ba4fcfadb8621896edfd271a9bd21 \
-                sha256 92bdeb4d1ba1a08f8a3aa1a0b10d36a06060c47f16fc523856f3ada598362020
+checksums       md5     120fc323649a6f83a755edbd052e9851 \
+                sha1    c60395360b7a527362e8149e68b12d364c4ee9eb \
```
Comment (pmetzger) [inline devel/zinc/Portfile]: You should remove the md5 and sha1 checksums and add rmd160.

File: devel/zinc/Portfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 PortSystem 1.0
 PortGroup       github 1.0
 
-github.setup    typesafehub zinc 0.3.5.3 v
+github.setup    typesafehub zinc 0.3.15 v
 categories      devel java
 license         Apache-2
 maintainers     openmaintainer {blair @blair}
```
Comment (pmetzger) [inline devel/zinc/Portfile]: Not your fault, but usually `openmaintainer` comes last on the line.

File: devel/zinc/Portfile (hunk 1 above)
Comment (l2dy) [reply devel/zinc/Portfile]: Some upstream provides only the weaker type of checksum, so I tend to keep them. My current understanding is that `sha256` and `size` are strongly recommended, and at least one other checksum is needed.

Comment (pmetzger) [reply devel/zinc/Portfile]: We always put in rmd160 these days, and `port -v checksum` will always report it to you.

Comment (pmetzger) [reply devel/zinc/Portfile]: (As the other checksums can be faked at this point there's no real point in keeping them, but I won't object overly much.)

Comment (blair) [reply devel/zinc/Portfile]: I like keeping the upstream checksums if they provide them.

Comment (blair) [reply devel/zinc/Portfile]: Yes, drop the md5 and sha1 as the upstream doesn't provide them and add the rmd160. I like providing the upstream checksums to make it easier for a reviewer to see that we haven't simply tried to substitute another tarball.

Comment (l2dy) [reply devel/zinc/Portfile]: @blair Done.

P.S. @pmetzger When there's no consensus, the port maintainer sets the rules.

**Comment by blair** [comment]
Thanks for the commit and making the changres!

File: devel/zinc/Portfile (hunk 1 above)
Comment (l2dy) [reply devel/zinc/Portfile]: > It's even been reified in the newest port lint (not yet released I think).

I think you are referring to https://github.com/macports/macports-base/commit/f32bca0e2d5a458ff2e4dcca65d9a155202b2841, which only warns about missing `size`, not other checksums.

Anyway, let's not discuss too much on a closed PR.

**Pull request merged**